# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1316724294 14400
#      Thu Sep 22 16:44:54 2011 -0400
# Node ID 968e89b45bbff3555ff170b1fb1b3b13ba7b9f95
# Parent  c5ddb99ff5de6e92f848663527c6147e526cdcc5
move tests for classes into subdirectories

* classes/test_classes.m: Rename from test_classes.m.
* classes/module.mk: New file with contents from @Blork/module.mk,
@Dork/module.mk, @Pork/module.mk, @Snork/module.mk, @Cork/module.mk,
@Gork/module.mk, @Sneetch/module.mk, and @Spork/module.mk.
(classes_FCN_FILES): Include test_classes.m in the list.
* @Blork/module.mk, @Dork/module.mk, @Pork/module.mk,
@Snork/module.mk, @Cork/module.mk, @Gork/module.mk,
@Sneetch/module.mk, @Spork/module.mk: Delete.

* ctor-vs-method/test_ctor_vs_method.m: Rename from test_ctor_vs_method.m.
* ctor-vs-method/module.mk (ctor_vs_method_FCN_FILES): Include
test_ctor_vs_method.m in the list.

* test/Makefile.am (FCN_FILES): Remove test_classes.m and
test_ctor_vs_method.m from the list.
Don't include @Blork/module.mk,
@Dork/module.mk, @Pork/module.mk, @Snork/module.mk, @Cork/module.mk,
@Gork/module.mk, @Sneetch/module.mk, @Spork/module.mk.
Do include classes/module.mk and ctor-vs-method/module.mk.

* fntests.m (run_test_dir): Also skip private directories and
directories with names that begin with "@".
Work recursively.  Change to test directory when running tests.

diff --git a/test/@Blork/module.mk b/test/@Blork/module.mk
deleted file mode 100644
--- a/test/@Blork/module.mk
+++ /dev/null
@@ -1,8 +0,0 @@
-at_Blork_FCN_FILES = \
-  @Blork/Blork.m \
-  @Blork/bleek.m \
-  @Blork/display.m \
-  @Blork/get.m \
-  @Blork/set.m
-
-FCN_FILES += $(at_Blork_FCN_FILES)
diff --git a/test/@Cork/module.mk b/test/@Cork/module.mk
deleted file mode 100644
--- a/test/@Cork/module.mk
+++ /dev/null
@@ -1,8 +0,0 @@
-at_Cork_FCN_FILES = \
-  @Cork/Cork.m \
-  @Cork/click.m \
-  @Cork/display.m \
-  @Cork/get.m \
-  @Cork/set.m
-
-FCN_FILES += $(at_Cork_FCN_FILES)
diff --git a/test/@Dork/module.mk b/test/@Dork/module.mk
deleted file mode 100644
--- a/test/@Dork/module.mk
+++ /dev/null
@@ -1,11 +0,0 @@
-at_Dork_FCN_FILES = \
-  @Dork/Dork.m \
-  @Dork/bling.m \
-  @Dork/display.m \
-  @Dork/gack.m \
-  @Dork/get.m \
-  @Dork/getStash.m \
-  @Dork/set.m \
-  @Dork/private/myStash.m
-
-FCN_FILES += $(at_Dork_FCN_FILES)
diff --git a/test/@Gork/module.mk b/test/@Gork/module.mk
deleted file mode 100644
--- a/test/@Gork/module.mk
+++ /dev/null
@@ -1,11 +0,0 @@
-at_Gork_FCN_FILES = \
-  @Gork/Gork.m \
-  @Gork/cork.m \
-  @Gork/display.m \
-  @Gork/gark.m \
-  @Gork/get.m \
-  @Gork/set.m \
-  @Gork/subsasgn.m \
-  @Gork/subsref.m
-
-FCN_FILES += $(at_Gork_FCN_FILES)
diff --git a/test/@Pork/module.mk b/test/@Pork/module.mk
deleted file mode 100644
--- a/test/@Pork/module.mk
+++ /dev/null
@@ -1,10 +0,0 @@
-at_Pork_FCN_FILES = \
-  @Pork/Pork.m \
-  @Pork/bling.m \
-  @Pork/display.m \
-  @Pork/get.m \
-  @Pork/gurk.m \
-  @Pork/set.m \
-  @Pork/private/myStash.m
-
-FCN_FILES += $(at_Pork_FCN_FILES)
diff --git a/test/@Sneetch/module.mk b/test/@Sneetch/module.mk
deleted file mode 100644
--- a/test/@Sneetch/module.mk
+++ /dev/null
@@ -1,5 +0,0 @@
-at_Sneetch_FCN_FILES = \
-  @Sneetch/Sneetch.m \
-  @Sneetch/display.m
-
-FCN_FILES += $(at_Sneetch_FCN_FILES)
diff --git a/test/@Snork/module.mk b/test/@Snork/module.mk
deleted file mode 100644
--- a/test/@Snork/module.mk
+++ /dev/null
@@ -1,17 +0,0 @@
-at_Snork_FCN_FILES = \
-  @Snork/Snork.m \
-  @Snork/cack.m \
-  @Snork/display.m \
-  @Snork/end.m \
-  @Snork/get.m \
-  @Snork/getStash.m \
-  @Snork/gick.m \
-  @Snork/loadobj.m \
-  @Snork/saveobj.m \
-  @Snork/set.m \
-  @Snork/subsasgn.m \
-  @Snork/subsindex.m \
-  @Snork/subsref.m \
-  @Snork/private/myStash.m
-
-FCN_FILES += $(at_Snork_FCN_FILES)
diff --git a/test/@Spork/module.mk b/test/@Spork/module.mk
deleted file mode 100644
--- a/test/@Spork/module.mk
+++ /dev/null
@@ -1,13 +0,0 @@
-at_Spork_FCN_FILES = \
-  @Spork/Spork.m \
-  @Spork/cack.m \
-  @Spork/display.m \
-  @Spork/geek.m \
-  @Spork/get.m \
-  @Spork/getStash.m \
-  @Spork/loadobj.m \
-  @Spork/saveobj.m \
-  @Spork/set.m \
-  @Spork/private/myStash.m
-
-FCN_FILES += $(at_Spork_FCN_FILES)
diff --git a/test/Makefile.am b/test/Makefile.am
--- a/test/Makefile.am
+++ b/test/Makefile.am
@@ -18,19 +18,17 @@
 # along with Octave; see the file COPYING.  If not, see
 # <http://www.gnu.org/licenses/>.
 
 include $(top_srcdir)/build-aux/common.mk
 
 FCN_FILES = \
   fntests.m \
   test_args.m \
-  test_classes.m \
   test_contin.m \
-  test_ctor_vs_method.m \
   test_diag_perm.m \
   test_error.m \
   test_eval-catch.m \
   test_for.m \
   test_func.m \
   test_global.m \
   test_if.m \
   test_index-wfi-f.m \
@@ -48,25 +46,17 @@ FCN_FILES = \
   test_struct.m \
   test_switch.m \
   test_system.m \
   test_transpose.m \
   test_try.m \
   test_unwind.m \
   test_while.m
 
-include @Blork/module.mk
-include @Cork/module.mk
-include @Dork/module.mk
-include @Gork/module.mk
-include @Pork/module.mk
-include @Sneetch/module.mk
-include @Snork/module.mk
-include @Spork/module.mk
-
+include classes/module.mk
 include ctor-vs-method/module.mk
 
 check: test_sparse.m test_bc_overloads.m
 	$(top_builddir)/run-octave --norc --silent --no-history $(srcdir)/fntests.m $(srcdir)
 
 test_sparse.m: build_sparse_tests.sh
 	$(srcdir)/build_sparse_tests.sh
 
diff --git a/test/@Blork/Blork.m b/test/classes/@Blork/Blork.m
rename from test/@Blork/Blork.m
rename to test/classes/@Blork/Blork.m
diff --git a/test/@Blork/bleek.m b/test/classes/@Blork/bleek.m
rename from test/@Blork/bleek.m
rename to test/classes/@Blork/bleek.m
diff --git a/test/@Blork/display.m b/test/classes/@Blork/display.m
rename from test/@Blork/display.m
rename to test/classes/@Blork/display.m
diff --git a/test/@Blork/get.m b/test/classes/@Blork/get.m
rename from test/@Blork/get.m
rename to test/classes/@Blork/get.m
diff --git a/test/@Blork/set.m b/test/classes/@Blork/set.m
rename from test/@Blork/set.m
rename to test/classes/@Blork/set.m
diff --git a/test/@Cork/Cork.m b/test/classes/@Cork/Cork.m
rename from test/@Cork/Cork.m
rename to test/classes/@Cork/Cork.m
diff --git a/test/@Cork/click.m b/test/classes/@Cork/click.m
rename from test/@Cork/click.m
rename to test/classes/@Cork/click.m
diff --git a/test/@Cork/display.m b/test/classes/@Cork/display.m
rename from test/@Cork/display.m
rename to test/classes/@Cork/display.m
diff --git a/test/@Cork/get.m b/test/classes/@Cork/get.m
rename from test/@Cork/get.m
rename to test/classes/@Cork/get.m
diff --git a/test/@Cork/set.m b/test/classes/@Cork/set.m
rename from test/@Cork/set.m
rename to test/classes/@Cork/set.m
diff --git a/test/@Dork/Dork.m b/test/classes/@Dork/Dork.m
rename from test/@Dork/Dork.m
rename to test/classes/@Dork/Dork.m
diff --git a/test/@Dork/bling.m b/test/classes/@Dork/bling.m
rename from test/@Dork/bling.m
rename to test/classes/@Dork/bling.m
diff --git a/test/@Dork/display.m b/test/classes/@Dork/display.m
rename from test/@Dork/display.m
rename to test/classes/@Dork/display.m
diff --git a/test/@Dork/gack.m b/test/classes/@Dork/gack.m
rename from test/@Dork/gack.m
rename to test/classes/@Dork/gack.m
diff --git a/test/@Dork/get.m b/test/classes/@Dork/get.m
rename from test/@Dork/get.m
rename to test/classes/@Dork/get.m
diff --git a/test/@Dork/getStash.m b/test/classes/@Dork/getStash.m
rename from test/@Dork/getStash.m
rename to test/classes/@Dork/getStash.m
diff --git a/test/@Dork/private/myStash.m b/test/classes/@Dork/private/myStash.m
rename from test/@Dork/private/myStash.m
rename to test/classes/@Dork/private/myStash.m
diff --git a/test/@Dork/set.m b/test/classes/@Dork/set.m
rename from test/@Dork/set.m
rename to test/classes/@Dork/set.m
diff --git a/test/@Gork/Gork.m b/test/classes/@Gork/Gork.m
rename from test/@Gork/Gork.m
rename to test/classes/@Gork/Gork.m
diff --git a/test/@Gork/cork.m b/test/classes/@Gork/cork.m
rename from test/@Gork/cork.m
rename to test/classes/@Gork/cork.m
diff --git a/test/@Gork/display.m b/test/classes/@Gork/display.m
rename from test/@Gork/display.m
rename to test/classes/@Gork/display.m
diff --git a/test/@Gork/gark.m b/test/classes/@Gork/gark.m
rename from test/@Gork/gark.m
rename to test/classes/@Gork/gark.m
diff --git a/test/@Gork/get.m b/test/classes/@Gork/get.m
rename from test/@Gork/get.m
rename to test/classes/@Gork/get.m
diff --git a/test/@Gork/set.m b/test/classes/@Gork/set.m
rename from test/@Gork/set.m
rename to test/classes/@Gork/set.m
diff --git a/test/@Gork/subsasgn.m b/test/classes/@Gork/subsasgn.m
rename from test/@Gork/subsasgn.m
rename to test/classes/@Gork/subsasgn.m
diff --git a/test/@Gork/subsref.m b/test/classes/@Gork/subsref.m
rename from test/@Gork/subsref.m
rename to test/classes/@Gork/subsref.m
diff --git a/test/@Pork/Pork.m b/test/classes/@Pork/Pork.m
rename from test/@Pork/Pork.m
rename to test/classes/@Pork/Pork.m
diff --git a/test/@Pork/bling.m b/test/classes/@Pork/bling.m
rename from test/@Pork/bling.m
rename to test/classes/@Pork/bling.m
diff --git a/test/@Pork/display.m b/test/classes/@Pork/display.m
rename from test/@Pork/display.m
rename to test/classes/@Pork/display.m
diff --git a/test/@Pork/get.m b/test/classes/@Pork/get.m
rename from test/@Pork/get.m
rename to test/classes/@Pork/get.m
diff --git a/test/@Pork/gurk.m b/test/classes/@Pork/gurk.m
rename from test/@Pork/gurk.m
rename to test/classes/@Pork/gurk.m
diff --git a/test/@Pork/private/myStash.m b/test/classes/@Pork/private/myStash.m
rename from test/@Pork/private/myStash.m
rename to test/classes/@Pork/private/myStash.m
diff --git a/test/@Pork/set.m b/test/classes/@Pork/set.m
rename from test/@Pork/set.m
rename to test/classes/@Pork/set.m
diff --git a/test/@Sneetch/Sneetch.m b/test/classes/@Sneetch/Sneetch.m
rename from test/@Sneetch/Sneetch.m
rename to test/classes/@Sneetch/Sneetch.m
diff --git a/test/@Sneetch/display.m b/test/classes/@Sneetch/display.m
rename from test/@Sneetch/display.m
rename to test/classes/@Sneetch/display.m
diff --git a/test/@Snork/Snork.m b/test/classes/@Snork/Snork.m
rename from test/@Snork/Snork.m
rename to test/classes/@Snork/Snork.m
diff --git a/test/@Snork/cack.m b/test/classes/@Snork/cack.m
rename from test/@Snork/cack.m
rename to test/classes/@Snork/cack.m
diff --git a/test/@Snork/display.m b/test/classes/@Snork/display.m
rename from test/@Snork/display.m
rename to test/classes/@Snork/display.m
diff --git a/test/@Snork/end.m b/test/classes/@Snork/end.m
rename from test/@Snork/end.m
rename to test/classes/@Snork/end.m
diff --git a/test/@Snork/get.m b/test/classes/@Snork/get.m
rename from test/@Snork/get.m
rename to test/classes/@Snork/get.m
diff --git a/test/@Snork/getStash.m b/test/classes/@Snork/getStash.m
rename from test/@Snork/getStash.m
rename to test/classes/@Snork/getStash.m
diff --git a/test/@Snork/gick.m b/test/classes/@Snork/gick.m
rename from test/@Snork/gick.m
rename to test/classes/@Snork/gick.m
diff --git a/test/@Snork/loadobj.m b/test/classes/@Snork/loadobj.m
rename from test/@Snork/loadobj.m
rename to test/classes/@Snork/loadobj.m
diff --git a/test/@Snork/private/myStash.m b/test/classes/@Snork/private/myStash.m
rename from test/@Snork/private/myStash.m
rename to test/classes/@Snork/private/myStash.m
diff --git a/test/@Snork/saveobj.m b/test/classes/@Snork/saveobj.m
rename from test/@Snork/saveobj.m
rename to test/classes/@Snork/saveobj.m
diff --git a/test/@Snork/set.m b/test/classes/@Snork/set.m
rename from test/@Snork/set.m
rename to test/classes/@Snork/set.m
diff --git a/test/@Snork/subsasgn.m b/test/classes/@Snork/subsasgn.m
rename from test/@Snork/subsasgn.m
rename to test/classes/@Snork/subsasgn.m
diff --git a/test/@Snork/subsindex.m b/test/classes/@Snork/subsindex.m
rename from test/@Snork/subsindex.m
rename to test/classes/@Snork/subsindex.m
diff --git a/test/@Snork/subsref.m b/test/classes/@Snork/subsref.m
rename from test/@Snork/subsref.m
rename to test/classes/@Snork/subsref.m
diff --git a/test/@Spork/Spork.m b/test/classes/@Spork/Spork.m
rename from test/@Spork/Spork.m
rename to test/classes/@Spork/Spork.m
diff --git a/test/@Spork/cack.m b/test/classes/@Spork/cack.m
rename from test/@Spork/cack.m
rename to test/classes/@Spork/cack.m
diff --git a/test/@Spork/display.m b/test/classes/@Spork/display.m
rename from test/@Spork/display.m
rename to test/classes/@Spork/display.m
diff --git a/test/@Spork/geek.m b/test/classes/@Spork/geek.m
rename from test/@Spork/geek.m
rename to test/classes/@Spork/geek.m
diff --git a/test/@Spork/get.m b/test/classes/@Spork/get.m
rename from test/@Spork/get.m
rename to test/classes/@Spork/get.m
diff --git a/test/@Spork/getStash.m b/test/classes/@Spork/getStash.m
rename from test/@Spork/getStash.m
rename to test/classes/@Spork/getStash.m
diff --git a/test/@Spork/loadobj.m b/test/classes/@Spork/loadobj.m
rename from test/@Spork/loadobj.m
rename to test/classes/@Spork/loadobj.m
diff --git a/test/@Spork/private/myStash.m b/test/classes/@Spork/private/myStash.m
rename from test/@Spork/private/myStash.m
rename to test/classes/@Spork/private/myStash.m
diff --git a/test/@Spork/saveobj.m b/test/classes/@Spork/saveobj.m
rename from test/@Spork/saveobj.m
rename to test/classes/@Spork/saveobj.m
diff --git a/test/@Spork/set.m b/test/classes/@Spork/set.m
rename from test/@Spork/set.m
rename to test/classes/@Spork/set.m
diff --git a/test/test_classes.m b/test/classes/test_classes.m
rename from test/test_classes.m
rename to test/classes/test_classes.m
diff --git a/test/ctor-vs-method/module.mk b/test/ctor-vs-method/module.mk
--- a/test/ctor-vs-method/module.mk
+++ b/test/ctor-vs-method/module.mk
@@ -1,10 +1,11 @@
 ctor_vs_method_FCN_FILES = \
   ctor-vs-method/@derived/derived.m \
   ctor-vs-method/@derived/parent.m \
   ctor-vs-method/@other/other.m \
   ctor-vs-method/@other/parent.m \
   ctor-vs-method/@parent/method.m \
   ctor-vs-method/@parent/parent.m \
-  ctor-vs-method/__trace__.m
+  ctor-vs-method/__trace__.m \
+  test_ctor_vs_method.m
 
 FCN_FILES += $(ctor_vs_method_FCN_FILES)
diff --git a/test/test_ctor_vs_method.m b/test/ctor-vs-method/test_ctor_vs_method.m
rename from test/test_ctor_vs_method.m
rename to test/ctor-vs-method/test_ctor_vs_method.m
diff --git a/test/fntests.m b/test/fntests.m
--- a/test/fntests.m
+++ b/test/fntests.m
@@ -110,36 +110,55 @@ endfunction
 
 function [dp, dn, dxf, dsk] = run_test_dir (fid, d);
   global files_with_tests;
   global files_with_no_tests;
   lst = dir (d);
   dp = dn = dxf = dsk = 0;
   for i = 1:length (lst)
     nm = lst(i).name;
-    if (length (nm) > 5 && strcmp (nm(1:5), "test_")
-        && strcmp (nm((end-1):end), ".m"))
-      p = n = xf = sk = 0;
-      ffnm = fullfile (d, nm);
-      if (has_tests (ffnm))
-        print_test_file_name (nm);
-        [p, n, xf, sk] = test (nm(1:(end-2)), "quiet", fid);
-        print_pass_fail (n, p);
-        files_with_tests(end+1) = ffnm;
-      ##elseif (has_demos (ffnm))
-      ##  files_with_tests(end+1) = ffnm;
-      else
-        files_with_no_tests(end+1) = ffnm;
-      endif
+    if (lst(i).isdir
+        && ! strcmp (nm, ".") && ! strcmp (nm, "..")
+        && ! strcmp (nm, "private") && nm(1) != "@"
+        && ! strcmp (nm, "CVS"))
+      [p, n, xf, sk] = run_test_dir (fid, [d, "/", nm]);
       dp += p;
       dn += n;
       dxf += xf;
       dsk += sk;
     endif
   endfor
+  saved_dir = pwd ();
+  unwind_protect
+    chdir (d);
+    for i = 1:length (lst)
+      nm = lst(i).name;
+      if (length (nm) > 5 && strcmp (nm(1:5), "test_")
+          && strcmp (nm((end-1):end), ".m"))
+        p = n = xf = sk = 0;
+        ffnm = fullfile (d, nm);
+        if (has_tests (ffnm))
+          print_test_file_name (nm);
+          [p, n, xf, sk] = test (nm(1:(end-2)), "quiet", fid);
+          print_pass_fail (n, p);
+          files_with_tests(end+1) = ffnm;
+        ##elseif (has_demos (ffnm))
+        ##  files_with_tests(end+1) = ffnm;
+        else
+          files_with_no_tests(end+1) = ffnm;
+        endif
+        dp += p;
+        dn += n;
+        dxf += xf;
+        dsk += sk;
+      endif
+    endfor
+  unwind_protect_cleanup
+    chdir (saved_dir);
+  end_unwind_protect
 endfunction
 
 function [dp, dn, dxf, dsk] = run_test_script (fid, d);
   global files_with_tests;
   global files_with_no_tests;
   global topsrcdir;
   global topbuilddir;
   lst = dir (d);
